### ISSUE TITLE
Add a Chinese language choice to the interface

### ISSUE THREAD

**Issue by ocrasborn** [Signbank/NGT-signbank#108]
...and thus generate a Chinese .po file
While doing this: Change flags to something more extendible: either a pull-down menu there on the right, or a menu 'Language' where both Content Language (issues to be created) and Interface Language can be changed.

**Comment by ocrasborn** [comment]
See http://unicode.org/faq/han_cjk.html on Chinese characters in unicode.
Does the server already have a font for Chinese?

**Comment by Woseseltops** [comment]
> Change flags to something more extendible: either a pull-down menu there on the right

I looked around for existing language picker pull downs, but using any of them would greatly sacrifice the minimalism we have now. I therefore created a quick design myself (left is normal state, right is when hovered):

![design language picker](https://cloud.githubusercontent.com/assets/1909317/11845668/4408dd08-a415-11e5-9afb-a97b44933ab4.png)

The flags change position when you click them. Shall I implement this? I expect it to take about an hour to get it right, but if I'm not mistaken this could also be a task for the interns next year.

**Comment by ocrasborn** [comment]
Please go ahead and implement it as you suggest (including the inclusion of Chinese options in the .po file of course): it remains to be seen whether we can find interns at all.

**Comment by Woseseltops** [comment]
I already have most of our new language picker locally (writing it as an independent generic thingy, so others can use it as well). I still need to:
* Add some subtile animations for a more professional look (if that's alright with you)
* Integrate it in Signbank

**Comment by ocrasborn** [comment]
@Woseseltops: quite alright :-)

**Comment by Woseseltops** [comment]
I integrated the new language picker this morning, and it's working, animations included! I tested it on various location of the site, both being logged in and logged out, both in Firefox and Chrome, and it now seems to be working everywhere.

As I said above, I made this a completely separate module in terms of functionality. Simply said, this means that (1) the available languages and (2) the Signbank logic that is executed when you actually pick a language, are not part of the language picker code. You can easily plug it into any other project.

Unfortunately, this does not go for the layout: I lost 2+ hours to get it in the menu bar without distorting both the menu bar and the picker, and finally the only solution I could come up with without losing the rest of the day was to change the language picker style sheet. Maybe one day I will give it another try to make things completely separate, but not in Signbank time as it won't benefit Signbank at all.

A few remarks / things to do:
  * If you change language when logged in, the page has to be reloaded two times (once to register your preference, once to actually change the site language); I want to try if I can make that more efficient.
  * The distances between flags are not quite right, and not always equal.
  * Don't ever pick Chinese, as it will put you in an infinite loop of the site trying to reload because the current language is not the language you picked, failing, and reloading as a result of that.

**Comment by Woseseltops** [comment]
Extra things noticed by @richardbank :
* Recently added signs are moved down when you hover the language picker flag
* At the detail view, the language picker disappears behind the bar below
* When the window is narrow, the flag appears on top of the name.

**Comment by Woseseltops** [comment]
Fixed:
* The distances between flags are not quite right, and not always equal.
* At the detail view, the language picker disappears behind the bar below
* Recently added signs are moved down when you hover the language picker flag

To do:
* If you change language when logged in, the page has to be reloaded two times
* When the window is narrow, the flag appears on top of the name.
* Don't ever pick Chinese

**Comment by ocrasborn** [comment]
I've made #62 top priority as well; this may be technically unrelated (or is it?), but it struck me again as something we really need to fix.

**Comment by Woseseltops** [comment]
Fixed:
* If you change language when logged in, the page has to be reloaded two times
* Don't ever pick Chinese

Now that you can pick Chinese, you can see that there's still some stuff to be done (besides all the translating :) )
* The page content and titles don't have a Chinese language option yet.
* The choice fields don't have a Chinese language option yet.
* There is some code that makes sure the correct choice field human value is shown, which has baked in that it can only be Dutch or English; that has to be made dynamic.

**Comment by Woseseltops** [comment]
Pfff... solved 10.000 database related problems, but I succeeded in adding the new chinese fields!

**Comment by ocrasborn** [comment]
That's a lot! But not as much as 10,000, luckily :-)

**Comment by ocrasborn** [comment]
So are we done with this, @Woseseltops ? (Where do we translate the menu items again? Are those all identical to the page titles?)

**Comment by Woseseltops** [comment]
I think so yes. If problems appear we'll reopen this issue.

**Comment by richardbank** [comment]
Something that hasn't been fixed yet:
* When the window is narrow, the flag appears on top of the name.

Moreover, something seems to be broken now, for when I hover over the flag, there is no fancy dropdown anymore, but flags appear in a horizontal row:
![image](https://cloud.githubusercontent.com/assets/10399929/12888623/58fd5d20-ce7b-11e5-890f-b081c6f9803a.png)
Still functions, though.

**Comment by Woseseltops** [comment]
Hmm... looks like the css folder I throw away in https://github.com/Signbank/NGT-signbank/commit/373346563cedc1e05adf8b7da0982342746b3db0 for issue #114 could not be completed removed. I put back the css file, and now it's working again!

**Comment by richardbank** [comment]
Well, yes, but 
* When the window is narrow, the flag appears on top of the name.

is still the case...

**Comment by Woseseltops** [comment]
I know, I know ;).

I decided, for now, on a quick and dirty solution, which is removing the overlap by moving the logout link to the left. This also means the link if further to the left when the screen is wide, but I thought that wouldn't be a problem; let me know if I'm wrong. By they way, if you make screen really narow, everything turns into a [hamburger menu](https://www.google.nl/search?q=hamburger+menu&espv=2&biw=1615&bih=963&source=lnms&tbm=isch&sa=X&ved=0ahUKEwi7zp2ipe3KAhWDrRoKHWLrDs8Q_AUIBigB&dpr=1#tbm=isch&q=hamburger+menu+icon). This had the same problem, so I also moved the hamburger to the left.

The real solution would be to have the header bar in 2-3 states, depending on screen size, and (re)move various items accordingly... but I'm not sure whether it would be worth the time.
